File: largest.rb
def largest_finder(list)
  largest = list[0]
  list.each do |num|
    if num > largest
      largest = num
    end
  end
  largest
end

def main
  number_list = [33, 65, 27, 223, 63]
  result = largest_finder(number_list)
  puts "the largest number is #{result}"
end

main

File: largest.ex
defmodule LargestFinder do
  def largest_finder([head | tail]) do
    Enum.reduce(tail, head, fn num, largest ->
      if num > largest do
        num
      else
        largest
      end
    end)
  end
end

defmodule Main do
  def main do
    number_list = [33, 65, 27, 223, 63]
    result = LargestFinder.largest_finder(number_list)
    IO.puts("the largest number is #{result}")
  end
end

Main.main()
